src/main/java/uk/gov/hmcts/reform/civil/service/docmosis/hearing/HearingFormGenerator.java

```diff
@@ -0,0 +1,124 @@
+                getFileName(caseData, template),
+                document.getBytes(),
+                DocumentType.HEARING_FORM
+            )
+        ));
+        return caseDocuments;
+    }
+
+    @Override
+    public HearingForm getTemplateData(CaseData caseData) {
+
+        return HearingForm.builder()
+            .court(caseData.getHearingLocation().getValue().getLabel())
+            .caseNumber(caseData.getLegacyCaseReference())
+            .creationDate(getDateFormatted(LocalDate.now()))
+            .claimant(caseData.getApplicant1().getPartyName())
+            .claimantReference(checkReference(caseData)
+                                   ? caseData.getSolicitorReferences().getApplicantSolicitor1Reference() : null)
+            .defendant(caseData.getRespondent1().getPartyName())
+            .defendantReference(checkReference(caseData)
+                                    ? caseData.getSolicitorReferences().getRespondentSolicitor1Reference() : null)
+            .hearingDate(getDateFormatted(caseData.getHearingDate()))
+            .hearingTime(getHearingTimeFormatted(caseData.getHearingTimeHourMinute()))
+            .hearingType(getHearingType(caseData))
+            .applicationDate(getDateFormatted(caseData.getDateOfApplication()))
+            .hearingDuration(getHearingDuration(caseData))
+            .additionalInfo(caseData.getInformation())
+            .feeAmount(HearingUtils.formatHearingFee(caseData.getHearingFee()))
+            .hearingDueDate(getDateFormatted(caseData.getHearingDueDate()))
+            .additionalText(caseData.getHearingNoticeListOther())
+            .claimant2exists(nonNull(caseData.getApplicant2()))
+            .defendant2exists(nonNull(caseData.getApplicant2()))
+            .claimant2(nonNull(caseData.getApplicant2()) ? caseData.getApplicant2().getPartyName() : null)
+            .defendant2(nonNull(caseData.getApplicant2()) ? caseData.getRespondent2().getPartyName() : null)
+            .claimant2Reference(checkReference(caseData)
+                                    ? caseData.getSolicitorReferences().getApplicantSolicitor1Reference() : null)
+            .defendant2Reference(checkReference(caseData)
+                                     ? caseData.getSolicitorReferences().getRespondentSolicitor2Reference() : null)
+            .build();
+    }
+
+    private String getFileName(CaseData caseData, DocmosisTemplates template) {
+        return String.format(template.getDocumentTitle(), caseData.getLegacyCaseReference());
+    }
+
+    private String getHearingTimeFormatted(String hearingTime) {
+        StringBuilder hearingTimeBuilder = new StringBuilder(hearingTime);
+        hearingTimeBuilder.insert(2, ':');
+        return hearingTimeBuilder.toString();
+    }
+
+    private String getDateFormatted(LocalDate date) {
+        if (isNull(date)) {
+            return null;
+        } else {
+            return date.format(DateTimeFormatter.ofPattern("dd/MMM/yyyy"));
```

Comment:
We appear to have a `DateFormatHelper` class. You might want to add a constant to that class for the new date format and then use the provided `DateFormatHelper::formatLocalDate` method.

I'm not pushed on using that method, but since we have it we should make sure that date formatting is handled consistently all throughout.

Also, if I'm reading Doug Keating's comments correctly, I believe this format is not among the ones he proposed... although I'm not familiar with Docmosis so I don't know how this would fit. On this part, please make sure the formatting matches the desired format and write a comment here to document the outcome.

Reply:
The format is defined in the Docmosis template, this format is so that Docmosis can read the date and format it